### PR TITLE
fix(vsts): Update build script to accept command line args

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -68,15 +68,15 @@ Function BuildProject($path, $message) {
     }
 }
 
-Function RunApp($path, $message, $args) {
+Function RunApp($path, $message, $params) {
```
Comment (abhipsaMisra) [inline build.ps1:71]: args is a keyword for passing unnamed args to a powershell script, so renamed it to "params" for our usage here.

File: build.ps1 [hunk 2]
```diff
@@ -103,26 +103,25 @@ try {
...
 
-        RunApp iot-hub\Samples\device\FileUploadSample "IoTHub\Device\FileUploadSample"
-        RunApp iot-hub\Samples\device\KeysRolloverSample "IoTHub\Device\KeysRolloverSample"
-        RunApp iot-hub\Samples\device\MessageSample "IoTHub\Device\MessageSample"
-        RunApp iot-hub\Samples\device\MethodSample "IoTHub\Device\MethodSample"
-        RunApp iot-hub\Samples\device\TwinSample "IoTHub\Device\TwinSample"
-
-        RunApp iot-hub\Samples\module\MessageSample "IoTHub\Module\MessageSample"
-        RunApp iot-hub\Samples\module\TwinSample "IoTHub\Module\TwinSample"
+        RunApp iot-hub\Samples\device\FileUploadSample "IoTHub\Device\FileUploadSample" "-p$env:IOTHUB_DEVICE_CONN_STRING"
```
Comment (abhipsaMisra) [inline build.ps1:110]: I got a weird error in powershell where sending the params as "-p $env:IOTHUB_DEVICE_CONN_STRING" introduced an extra whitespace before the value of `$env:IOTHUB_DEVICE_CONN_STRING`. Adding braces/ quotes around `$env:IOTHUB_DEVICE_CONN_STRING` didn't help either.

The only thing that worked was to send "-p$env:IOTHUB_DEVICE_CONN_STRING" (without the whitespace delimiter), but this seems inherently wrong to me.

Any ideas?

Comment (drwill-ms) [reply build.ps1:110]: Yeah, this is going to look like a mistake to any future observer.

Have you tried `$($env.IOTHUB_DEVICE_CONN_STRING)`? You might have to do that if encasing it in quotes, anyway. That is how you embed a hierarchical variable in a string in powershell.

If nothing else works, then put a comment in to explain why there is no whitespace.

File: build.ps1 [hunk 3]
```diff
@@ -103,26 +103,25 @@ try {
...
+        RunApp iot-hub\Samples\device\MessageReceiveSample "IoTHub\Device\MessageReceiveSample" "-p$env:IOTHUB_DEVICE_CONN_STRING"
+        RunApp iot-hub\Samples\device\MethodSample "IoTHub\Device\MethodSample" "-p$env:IOTHUB_DEVICE_CONN_STRING"
+        RunApp iot-hub\Samples\device\TwinSample "IoTHub\Device\TwinSample" "-p$env:IOTHUB_DEVICE_CONN_STRING"
+        # DeviceStreaming sample is not added since it is not available in all regions.
+        # DeviceReconnectionSample is not added since it is designed to run until cancelled using Console input.
```
Comment (drwill-ms) [inline build.ps1:115]: You could add a parameter to the sample for run duration too.

Comment (drwill-ms) [reply build.ps1:115]: Then the cancellationToken signaled by the control-C would conditionally have a timespan associated with it.

**Review by drwill-ms — APPROVED** [review]

**Review by azabbasi — APPROVED** [review]

File: build.ps1 (hunk 3 above)
Comment (abhipsaMisra) [reply build.ps1:115]: I can give that a try, sure

File: build.ps1 (hunk 2 above)
Comment (abhipsaMisra) [reply build.ps1:110]: `RunApp iot-hub\Samples\device\FileUploadSample "IoTHub\Device\FileUploadSample" "-p $($env:IOTHUB_DEVICE_CONN_STRING)"` also introduced the space 😞 
I'll add a note explaining this workaround for now.

@azabbasi - you can update it with your findings when you have the time :)

Comment (abhipsaMisra) [reply build.ps1]: @drwill-ms ,@azabbasi - This is now solved, just FYI

File: build.ps1 [hunk 4]
```diff
@@ -68,15 +68,16 @@ Function BuildProject($path, $message) {
...
-    $label = "RUN: --- $message $configuration ($args)---"
+    $label = "RUN: --- $message $configuration ($params)---"
 
     Write-Host
     Write-Host -ForegroundColor Cyan $label
     cd (Join-Path $rootDir $path)
 
-    & dotnet run $args
+    $runCommand = "dotnet run -- $params"
+    Invoke-Expression $runCommand
```
Comment (abhipsaMisra) [inline build.ps1:80]: The native dotnet run command when invoked using `&` was showing this weird behavior, invoking it using `Invoke-Expression` works just fine.

File: build.ps1 (hunk 2 above)
Comment (drwill-ms) [reply build.ps1]: double quotes eh?

Comment (abhipsaMisra) [reply build.ps1]: Actually it was related to the way the command was being invoked. Invoking with `&` gives rise to this issue, whereas invoking with `Invoke-Expression` works as expected.

File: iot-hub/Samples/device/MessageReceiveSample/MessageReceiveSample.cs [hunk 5]
```diff
@@ -71,8 +71,6 @@ private async Task ReceiveC2dMessagesPollingAndComplete(TimeSpan timeout)
 
                 await _deviceClient.CompleteAsync(receivedMessage);
                 Console.WriteLine($"{DateTime.Now}> Completed C2D message with Id={receivedMessage.MessageId}.");
-
-                receivedMessage.Dispose();
```
Comment (abhipsaMisra) [inline iot-hub/Samples/device/MessageReceiveSample/MessageReceiveSample.cs:75]: initialized using `using Message receivedMessage = await _deviceClient.ReceiveAsync(timeout);`, so will get disposed automatically once out-of-scope

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**